share/doc/homebrew/Node-for-Formula-Authors.md

```diff
@@ -0,0 +1,115 @@
+# Node for formula authors
+
+This document explains how to successfully use Node and npm in a node module based Homebrew formula.
+
+# Running `npm install`
+
+Homebrew provides two helper methods in a `Language::Node` module, `std_npm_install_args` and `local_npm_install_args`. They both set up the correct environment for npm and return arguments for `npm install` for their specific use cases. Please use them instead of invoking `npm install` explicitly. The syntax for a standard node module installation is:
+
+```ruby
+system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+```
+
+where `libexec` is the destination prefix (usually the `libexec` variable).
+
+# Download URL
+
+If the node module is also available on the npm registry, we prefer npm hosted release tarballs over github (or elsewhere) hosted source tarballs. The advantages of these tarballs are that they doesn't include the files from the `.npmignore` (such as tests) resulting in a smaller download size and that a possibly transpilation step is already done (e.g. no need to compile Coffee Script files as a build step).
```

Comment:
You're going to want to defenestrate me by the end of this review I'm sure, but it should be GitHub rather than `github`.